### PR TITLE
modified versions of KB and K8S

### PR DESCRIPTION
**Pull request by laxmikantbpandhare**
Bump k8s and KB dependencies in java-operator-plugins

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -7,6 +7,6 @@ require github.com/spf13/pflag v1.0.5
 require (
 	github.com/onsi/ginkgo v1.15.2
 	github.com/onsi/gomega v1.11.0
-	k8s.io/apimachinery v0.20.2
-	sigs.k8s.io/kubebuilder/v3 v3.0.0-beta.1
+	k8s.io/apimachinery v0.21.2
+	sigs.k8s.io/kubebuilder/v3 v3.1.0
```
Comment (varshaprasad96) [inline go.mod:11]: can you bump this to the latest commit of kb https://github.com/kubernetes-sigs/kubebuilder/commit/742983631190812ce81402fb662998fe07447e21

Comment (varshaprasad96) [reply go.mod:11]: After you bump this, ginkgo and gomega versions would also be bumped.

**Review by varshaprasad96 — CHANGES_REQUESTED** [review]

**Comment by varshaprasad96** [comment]
@laxmikantbpandhare after you bump these dependencies, can you run `go mod tidy`.

**Comment by varshaprasad96** [comment]
Also, can we bump go to 1.16?
cc: @jmrodri

File: go.mod (hunk 1 above)
Comment (laxmikantbpandhare) [reply go.mod]: Yes Modified it.

Comment (laxmikantbpandhare) [reply go.mod]: ginko and gomega version also got updated.

**Comment by laxmikantbpandhare** [comment]
> @laxmikantbpandhare after you bump these dependencies, can you run `go mod tidy`.

Yes, I did this.

**Review by varshaprasad96 — APPROVED** [review]

**Comment by varshaprasad96** [comment]
@laxmikantbpandhare I'm not sure if the CI has e2e tests. Can you verify if everything works as expected after the version bump. Just for a final verification before merging the PR, since C-R 0.9+ has quite a few breaking changes.

**Comment by laxmikantbpandhare** [comment]
> @laxmikantbpandhare I'm not sure if the CI has e2e tests. Can you verify if everything works as expected after the version bump. Just for a final verification before merging the PR, since C-R 0.9+ has quite a few breaking changes.

I verified using memcached operator. I was able to scaffold an operator using init and create api command. The operator is compiled and ran successfully.

**Pull request merged**